functions/_pure_prompt_symbol.fish

```diff
@@ -14,5 +17,18 @@ function _pure_prompt_symbol \
         end
     end
 
-    echo "$color_symbol$pure_symbol"
+    if test reverse_symbol_instead_of_default_mode_prompt
```

Comment:
This is a code smell, better write it like:

1. set `$pure_symbol` to `❯` 
1. change `$pure_symbol` to `❮` if `$reverse_symbol_instead_of_default_mode_prompt` is `true`
1. concatenate `$color_symbol` with `$pure_symbol`

In a latter iteration, step 2's logic should be extracted to its own file `functions/_pure_reverse_prompt_symbol_in_vimode.fish`